interfaces/builtin/joystick.go

```diff
@@ -0,0 +1,95 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/snap"
+)
+
+const joystickConnectedPlugAppArmor = `
```

Comment:
Silly(?) question - but given that it appears this is only having an apparmor snippet and is otherwise a very simple interface - would it make sense to just use the `commonInterface` abstraction/helper here? Similar to e.g. `camera.go`. This way joystick.go could probably be written as:
```
const joystickConnectedPlugAppArmor = `
 /dev/input/js[0-9]* rw, 
 /run/udev/data/c13:{[0-9],[12][0-9],3[01]} r, 
`
func NewJoystickInterface() interfaces.Interface {
	return &commonInterface{
		name: "joystick",
		connectedPlugAppArmor: joystickConnectedPlugAppArmor,
		reservedForOS:         true,
	}
}
```
I put a better diff here http://paste.ubuntu.com/24305321/